en/02_Developer_Guides/02_Controllers/07_CMS_JSON_APIs.md

```diff
@@ -0,0 +1,154 @@
+---
+title: CMS JSON APIs
+summary: Creating standardised JSON APIs for authenticated users in the CMS.
+---
+
+# CMS JSON APIs
+
+This document contains a standard set of conventions to be used when creating JSON APIs in the CMS that are used in conjunction with AJAX requests from authenticated CMS users.
+
+To view an example of a controller that follows these standards see [`LinkFieldController`](https://github.com/silverstripe/silverstripe-linkfield/blob/4/src/Controllers/LinkFieldController.php).
+
+## Making the controller "REST-like" and its relation with `FormSchema`
+
+It's recommend you design your API with a "REST-like" approach for JSON requests in your application, acknowledging that certain aspects may not strictly adhere to pure REST principles. This means that you should use the HTTP methods GET, POST, and DELETE, though not others such as PUT or PATCH.
+
+The reason for not implementing PUT or PATCH is because `FormSchema` and react `FormBuilder` should be used whenever you have need to render a form that uses the standard set of react form fields. `FormSchema` and `FormBuilder` will handle data submission, data validation and showing validation errors on the frontend.
+
+FormSchema diverges from REST principles as it utilises a combination of JSON to retrieve the `FormSchema` and submitting data using `application/x-www-form-urlencoded` POST requests. This method has proven effective, and there are currently no plans to alter its functionality.
+
+Becaues of this you should avoid updating large parts of a DataObject with either a POST, PUT or PATCH requests as you should be using `FormSchema` to update DataObjects. Only use POST requests to submit small amounts of data such as creating a new record without any data and updating sort order fields.
```

Comment:
```suggestion
Because of this you should avoid updating large parts of a DataObject with either a POST, PUT or PATCH requests as you should be using `FormSchema` to update DataObjects. Only use POST requests to submit small amounts of data such as creating a new record without any data or updating sort order fields.
```

Reply:
Updated